### PR TITLE
Prd 2729 support bdm ocean k8s aws cluster

### PR DESCRIPTION
**Pull request by sharadkesarwani**
prd-2729 support bdm ocean k8s aws cluster.

# Jira Ticket
Ref: https://spotinst.atlassian.net/browse/SPOTAUT-12057

### CODE REVIEW

File: service/ocean/providers/aws/cluster.go [hunk 1]
```diff
@@ -2030,3 +2031,184 @@ func (o *Filters) SetVirtualizationTypes(v []string) *Filters {
...
+func (o *LaunchSpecification) SetClusterBlockDeviceMappings(v []*ClusterBlockDeviceMapping) *LaunchSpecification {
+	if o.BlockDeviceMappings = v; o.BlockDeviceMappings == nil {
+		o.nullFields = append(o.nullFields, "BlockDeviceMappings")
+	}
+	return o
+}
+
+type ClusterBlockDeviceMapping struct {
+	DeviceName  *string     `json:"deviceName,omitempty"`
+	NoDevice    *string     `json:"noDevice,omitempty"`
```
Comment (chandra1-n) [inline service/ocean/providers/aws/cluster.go:2044]: NoDevice and VirtualName not available in the configuration in documentation

File: service/ocean/providers/aws/cluster.go [hunk 2]
```diff
@@ -2030,3 +2031,184 @@ func (o *Filters) SetVirtualizationTypes(v []string) *Filters {
...
+}
+
+func (o *ClusterBlockDeviceMapping) SetDeviceName(v *string) *ClusterBlockDeviceMapping {
+	if o.DeviceName = v; o.DeviceName == nil {
+		o.nullFields = append(o.nullFields, "DeviceName")
+	}
+	return o
+}
+
+func (o *ClusterBlockDeviceMapping) SetNoDevice(v *string) *ClusterBlockDeviceMapping {
```
Comment (chandra1-n) [inline service/ocean/providers/aws/cluster.go:2089]: recheck the documentation for this param

File: service/ocean/providers/aws/cluster.go [hunk 3]
```diff
@@ -2030,3 +2031,184 @@ func (o *Filters) SetVirtualizationTypes(v []string) *Filters {
...
+	if o.NoDevice = v; o.NoDevice == nil {
+		o.nullFields = append(o.nullFields, "NoDevice")
+	}
+	return o
+}
+
+func (o *ClusterBlockDeviceMapping) SetVirtualName(v *string) *ClusterBlockDeviceMapping {
```
Comment (chandra1-n) [inline service/ocean/providers/aws/cluster.go:2096]: recheck the documentation for this param

File: service/ocean/providers/aws/cluster.go [hunk 4]
```diff
@@ -162,21 +162,22 @@ type Filters struct {
...
+	UserData                 *string                      `json:"userData,omitempty"`
+	IAMInstanceProfile       *IAMInstanceProfile          `json:"iamInstanceProfile,omitempty"`
+	Tags                     []*Tag                       `json:"tags,omitempty"`
+	LoadBalancers            []*LoadBalancer              `json:"loadBalancers,omitempty"`
+	RootVolumeSize           *int                         `json:"rootVolumeSize,omitempty"`
+	Monitoring               *bool                        `json:"monitoring,omitempty"`
+	EBSOptimized             *bool                        `json:"ebsOptimized,omitempty"`
+	UseAsTemplateOnly        *bool                        `json:"useAsTemplateOnly,omitempty"`
+	InstanceMetadataOptions  *InstanceMetadataOptions     `json:"instanceMetadataOptions,omitempty"`
+	BlockDeviceMappings      []*ClusterBlockDeviceMapping `json:"blockDeviceMappings,omitempty"`
```
Comment (chandra1-n) [inline service/ocean/providers/aws/cluster.go:179]: change to ClusterBlockDeviceMappings

**Review by chandra1-n — CHANGES_REQUESTED** [review]
please fix the comments

File: service/ocean/providers/aws/cluster.go (hunk 4 above)
Comment (sharadkesarwani) [reply service/ocean/providers/aws/cluster.go]: Update name to ClusterBlockDeviceMappings

File: service/ocean/providers/aws/cluster.go (hunk 3 above)
Comment (sharadkesarwani) [reply service/ocean/providers/aws/cluster.go]: removed unused parameters VirtualName.

File: service/ocean/providers/aws/cluster.go (hunk 1 above)
Comment (sharadkesarwani) [reply service/ocean/providers/aws/cluster.go]: removed unused parameters VirtualName and NoDevice

**Review by chandra1-n — APPROVED** [review]

**Pull request merged**